railties/lib/rails/migration_controller.rb

```diff
@@ -0,0 +1,15 @@
+require "rails/application_controller"
+
+class Rails::MigrationController < Rails::ApplicationController # :nodoc:
+  before_action :require_local!
+
+  def migrate
+    require 'rake'
+    Rake::Task.clear
+    Rails.application.load_tasks
+    Rake::Task['db:migrate'].invoke
```

Comment:
Since I just merged https://github.com/rails/rails/pull/26414, we should be able to do this instead:

```ruby
# Alongside the other require...
require "rails/command"

# Later...
def migrate
  Rails::Command.invoke("db:migrate")

  redirect_to "/"
end
```